src/helm_client.rs

```diff
@@ -0,0 +1,181 @@
+        }
+
+        // If checks succeed, create Helm client
+        Ok(Self {})
+    }
+
+    /// Installs the given chart under the given name.
+    ///
+    #[instrument(skip(self))]
+    pub fn install(&self, args: &InstallArg) -> Result<(), HelmError> {
+        let mut command = args.install();
+        command.result()?;
+        Ok(())
+    }
+
+    /// Upgrades the given chart
+    #[instrument(skip(self))]
+    pub fn upgrade(&self, args: &InstallArg) -> Result<(), HelmError> {
+        let mut command = args.upgrade();
+        command.result()?;
+        Ok(())
+    }
+
+    /// Uninstalls specified chart library
+    pub fn uninstall(&self, uninstall: UninstallArg) -> Result<(), HelmError> {
+        if uninstall.ignore_not_found {
+            let app_charts = self
+                .get_installed_chart_by_name(&uninstall.release, uninstall.namespace.as_deref())?;
+            if app_charts.is_empty() {
+                warn!("Chart does not exists, {}", &uninstall.release);
+                return Ok(());
+            }
+        }
+        let mut command: Command = uninstall.into();
+        command.result()?;
+        Ok(())
+    }
+
+    /// Adds a new helm repo with the given chart name and chart location
+    #[instrument(skip(self))]
+    pub fn repo_add(&self, chart: &str, location: &str) -> Result<(), HelmError> {
+        Command::new("helm")
+            .args(&["repo", "add", chart, location])
+            .result()?;
+        Ok(())
+    }
+
+    /// Updates the local helm repository
+    #[instrument(skip(self))]
+    pub fn repo_update(&self) -> Result<(), HelmError> {
+        Command::new("helm").args(&["repo", "update"]).result()?;
+        Ok(())
+    }
+
+    /// Searches the repo for the named helm chart
+    #[instrument(skip(self))]
+    pub fn search_repo(&self, chart: &str, version: &str) -> Result<Vec<Chart>, HelmError> {
+        let mut command = Command::new("helm");
+        command
+            .args(&["search", "repo", chart])
+            .args(&["--version", version])
+            .args(&["--output", "json"]);
+
+        let output = command.result()?;
+
+        check_helm_stderr(output.stderr)?;
+        serde_json::from_slice(&output.stdout).map_err(HelmError::Serde)
+    }
+
+    /// Get all the available versions
+    #[instrument(skip(self))]
+    pub fn versions(&self, chart: &str) -> Result<Vec<Chart>, HelmError> {
+        let mut command = Command::new("helm");
+        command
+            .args(&["search", "repo"])
+            .args(&["--versions", chart])
+            .args(&["--output", "json", "--devel"]);
+        let output = command.result()?;
+
+        check_helm_stderr(output.stderr)?;
+        serde_json::from_slice(&output.stdout).map_err(HelmError::Serde)
+    }
+
+    /// Checks that a given version of a given chart exists in the repo.
+    #[instrument(skip(self))]
+    pub fn chart_version_exists(&self, name: &str, version: &str) -> Result<bool, HelmError> {
```

Comment:
similar to repo, I think we should move all chart related commands to `Chart` object

Reply:
https://github.com/infinyon/fluvio-helm/issues/20